code/mob/living/critter/small_animal.dm

```diff
@@ -513,17 +508,9 @@ proc/filter_carrier_pets(var/type)
 				boutput(M, SPAN_ALERT("You feel a wave of sadness wash over you, something terrible has happened."))
 		src.UpdateIcon()
 
-	on_pet(mob/user)
-		if (..())
-			return 1
-		if (src.ai?.enabled && ishuman(user))
-			var/mob/living/carbon/human/human = user
-			var/clown_tally = human.clown_tally()
-			if (clown_tally>=2 || human.traitHolder.hasTrait("training_clown"))
-				src.ai.priority_tasks += src.ai.get_instance(/datum/aiTask/sequence/goalbased/critter/attack, list(src, src.ai.default_task))
-				src.ai.interrupt()
-				src.visible_message(SPAN_NOTICE("[src] notices a Clown and knocks [human] over!"))
-				human.setStatus("resting", duration = INFINITE_STATUS)
+	updatehealth()
```

Comment:
Don't put this here, calling `UpdateIcon` every time something takes damage is wasteful. Why not just put it in `death` and `full_heal`?